### ISSUE TITLE
MATCH function fails to locate integer values in csv.

### ISSUE THREAD

**Issue by adjenks** [PHPOffice/PhpSpreadsheet#1232]
This is:

```
- [x] a bug report
- [ ] a feature request
- [ ] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected behavior?

Return a match when using MATCH to find integer values.

### What is the current behavior?

Returns `#N/A`

### What are the steps to reproduce?

The `MATCH()` function broke for me because I am looking up an int value but value types are compared internally and are all cast to float.
The comparison is done here:

https://github.com/PHPOffice/PhpSpreadsheet/blob/9df68f12e2ebe2987ceee8bdfe1a04b8c3fe5691/src/PhpSpreadsheet/Calculation/LookupRef.php#L529

Random relevant quote I found regarding this error: 
> MS Excel doesn't differentiate between integer and float, it just has a type number; so internally PHPSpreadsheet uses a float for numbers
https://github.com/PHPOffice/PhpSpreadsheet/issues/318#issuecomment-469909608

As an example:
```php
<?php

require __DIR__ . '/vendor/autoload.php';

// Create new Spreadsheet object
$spreadsheet = \PhpOffice\PhpSpreadsheet\IOFactory::load('./testsheet.csv');
$sheet1 = $spreadsheet->getSheet(0);
var_export($sheet1->setCellValue('AA99', '=MATCH(1000,A:A)')
	->getCell('AA99')
	->getCalculatedValue()
);
die();
```

Accompany this test with a sheet containing values like so:
```csv
1000,5,5,5
2000,5,5,5
```

In the above example, `MATCH()` will attempt to find 1000, however the spreadsheet value is cast to a float and becomes 1000.0 but the value passed into the match function remains an integer and the type comparison fails. This happened to me when importing the sheet as a csv.

This deviates from behavior in excel, excel will match with this formula: `=MATCH(1000,A:A)`

As a workaround in PhpSpreadsheet, the `$lookupValue` can be appended with a `.0` to make it a float (`=MATCH(1000.0,A:A)`). This is what I will do in the meantime.

### Which versions of PhpSpreadsheet and PHP are affected?

At least 1.9.0, didn't check others. Appears to have happened in the last modification to the LookupRef.php file.

**Comment by adjenks** [comment]
*ACTIVITY*

**Comment by adjenks** [comment]
jkhlkjhlkjhkljhkljhgiu97fg9fp68d86dfp8y6pf6y

**Comment by adjenks** [comment]
OMFG you stupid bot.

**Comment by arnested** [comment]
This sounds like what I just fixed in #1521.

**Comment by adjenks** [comment]
@arnested Awesome. Good work.
